### PR TITLE
Unit tests for the extract scripts: demand & generation

### PR DESCRIPTION
**Pull request by JoshuaMarden**
Add unit testing for the demand and generation scrips.

### CODE REVIEW

**Comment by JoshuaMarden** [comment]
Sorry for the conflicts.

**Comment by JoshuaMarden** [comment]
Fixed conflicts.

File: pipeline/dockerfile
Comment (HighestAuto) [inline pipeline/dockerfile]: Is this file needed?

File: pipeline/extract_to_s3.py
Comment (HighestAuto) [inline pipeline/extract_to_s3.py]: This will break the code if pushed as an image to the Cloud, Does this need to be here?

File: pipeline/extract_to_s3.py
Comment (HighestAuto) [inline pipeline/extract_to_s3.py]: There is no need to import extract_carbon

File: pipeline/load.py
Comment (HighestAuto) [inline pipeline/load.py]: We can also delete this file

**Review by HighestAuto — COMMENTED** [review]
There are a few files that you need to fix before approval, mainly the need for pipeline.extract*

File: tests/extract_demand_test.py
Comment (HighestAuto) [inline tests/extract_demand_test.py:2]: Could you add these tests to a class to sort them by what they are testing? honestly a preference but makes the tests more readable

**Review by HighestAuto — APPROVED** [review]
Okay looking good! These changes help keep the integrity of our code to a high standard

**Pull request merged**
